sklearn/tree/_export.py

```diff
@@ -683,19 +687,24 @@ def export(self, decision_tree, ax=None):
     def recurse(self, node, tree, ax, max_x, max_y, depth=0):
         import matplotlib.pyplot as plt
 
-        kwargs = dict(
-            bbox=self.bbox_args.copy(),
+        # kwargs for annotations without a bounding box
+        non_box_kwargs = dict(
             ha="center",
             va="center",
             zorder=100 - 10 * depth,
             xycoords="axes fraction",
+        )
+        if self.fontsize is not None:
+            non_box_kwargs["fontsize"] = self.fontsize
+
+        # kwargs for annotations with a bounding box
+        kwargs = dict(
```

Comment:
```suggestion
        kwargs = dict(
            ha="center",
```

We add `ha="center"` here. Perhaps you missed this part so that you got that broken version of graph.

Reply:
Ah I see. Yes that is I think the issue that was causing the weird graphs. I confirm your suggestions are much more robust.